SU2_CFD/src/python_wrapper_structure.cpp

```diff
@@ -914,116 +616,34 @@ void CSinglezoneDriver::SetInitialMesh() {
 
-#ifdef HAVE_MPI
-  MPI_Comm_rank(SU2_MPI::GetComm(), &rank);
-#endif
+  SU2_MPI::Comm_rank(SU2_MPI::GetComm(), &rank);
 
   if(rank == MASTER_NODE) cout << "Updating boundary conditions." << endl;
   for(iZone = 0; iZone < nZone; iZone++){
     geometry_container[iZone][INST_0][MESH_0]->UpdateCustomBoundaryConditions(geometry_container[iZone][INST_0], config_container[iZone]);
   }
 }
 
-void CDriver::SetMeshDisplacement(unsigned short iMarker, unsigned long iVertex, passivedouble DispX, passivedouble DispY, passivedouble DispZ) {
-
-  unsigned long iPoint;
-  PyWrapVarCoord[0] = DispX;
-  PyWrapVarCoord[1] = DispY;
-  PyWrapVarCoord[2] = DispZ;
-
-  iPoint = geometry_container[ZONE_0][INST_0][MESH_0]->vertex[iMarker][iVertex]->GetNode();
-
-  solver_container[ZONE_0][INST_0][MESH_0][MESH_SOL]->GetNodes()->SetBound_Disp(iPoint,PyWrapVarCoord);
-
-}
-
-void CDriver::CommunicateMeshDisplacement(void) {
-
-  solver_container[ZONE_0][INST_0][MESH_0][MESH_SOL]->InitiateComms(geometry_container[ZONE_0][INST_0][MESH_0],
-                                                                    config_container[ZONE_0], MESH_DISPLACEMENTS);
-  solver_container[ZONE_0][INST_0][MESH_0][MESH_SOL]->CompleteComms(geometry_container[ZONE_0][INST_0][MESH_0],
-                                                                    config_container[ZONE_0], MESH_DISPLACEMENTS);
-
-}
-
-vector<passivedouble> CDriver::GetMeshDisp_Sensitivity(unsigned short iMarker, unsigned long iVertex) {
-
-  unsigned long iPoint;
-  vector<su2double> Disp_Sens(3, 0.0);
-  vector<passivedouble> Disp_Sens_passive(3, 0.0);
-
-  iPoint = geometry_container[ZONE_0][INST_0][MESH_0]->vertex[iMarker][iVertex]->GetNode();
-  CSolver *solver =  solver_container[ZONE_0][INST_0][MESH_0][ADJMESH_SOL];
-  CGeometry *geometry = geometry_container[ZONE_0][INST_0][MESH_0];
-
-  Disp_Sens[0] = solver->GetNodes()->GetBoundDisp_Sens(iPoint, 0);
-  Disp_Sens[1] = solver->GetNodes()->GetBoundDisp_Sens(iPoint, 1);
-  if (geometry->GetnDim() == 3)
-    Disp_Sens[2] = solver->GetNodes()->GetBoundDisp_Sens(iPoint, 2);
-  else
-    Disp_Sens[2] = 0.0;
-
-  Disp_Sens_passive[0] = SU2_TYPE::GetValue(Disp_Sens[0]);
-  Disp_Sens_passive[1] = SU2_TYPE::GetValue(Disp_Sens[1]);
-  Disp_Sens_passive[2] = SU2_TYPE::GetValue(Disp_Sens[2]);
-
-  return Disp_Sens_passive;
-
-}
+////////////////////////////////////////////////////////////////////////////////
+/* Functions related to finite elements                                       */
+////////////////////////////////////////////////////////////////////////////////
 
 void CDriver::SetFEA_Loads(unsigned short iMarker, unsigned long iVertex, passivedouble LoadX,
                        passivedouble LoadY, passivedouble LoadZ) {
 
   unsigned long iPoint;
-  PyWrapNodalForce[0] = LoadX;
-  PyWrapNodalForce[1] = LoadY;
-  PyWrapNodalForce[2] = LoadZ;
+  vector<su2double> NodalForce (3,0.0);
+  NodalForce[0] = LoadX;
+  NodalForce[1] = LoadY;
+  NodalForce[2] = LoadZ;
```

Comment:
The pattern in these functions is to load the 3 separate components into an array or vector to then pass by pointer to the solver.
Since we know the number of components (from the function signature) it is better to use a fixed-size temporary for that, e.g. su2double NodalForce[3]; or array<su2double,3> NodalForce, since that avoids dynamic allocation when this function is called for each vertex.
For the functions that return small vectors we would ideally do the same, return a fixed size type (the only choice there would be array<su2double,3>) but maybe python/swig has some limitation.

Reply:
Thank you very much for the comment!
I do not think swig is limited in that sense, the limitation was mine not perfectly understanding the difference, I see now.
I am changing it